contracts/icp/context-config/src/types.rs

```diff
@@ -0,0 +1,320 @@
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+use bs58::decode::Result as Bs58Result;
+use calimero_context_config::repr::{self, LengthMismatch, Repr, ReprBytes, ReprTransmute};
+use calimero_context_config::types::{
+    Application, ApplicationMetadata, ApplicationSource, Capability,
+};
+use candid::CandidType;
+use ed25519_dalek::{Verifier, VerifyingKey};
+use serde::{Deserialize, Serialize};
+
+use crate::guard::Guard;
+
+#[derive(
+    CandidType, Serialize, Deserialize, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Hash,
+)]
+pub struct Identity(pub [u8; 32]);
```

Comment:
these (shouldn't) need to be pub

```suggestion
pub struct Identity([u8; 32]);
```

you can check to be sure, but anything that needs to access the contained data should call `.as_bytes` instead